vaadin-grid-flow-parent/vaadin-grid-flow-integration-tests/src/test/java/com/vaadin/flow/component/grid/it/GridViewIT.java

```diff
@@ -693,6 +693,23 @@ public void gridWithFiltering() {
         waitUntil(driver -> grid.getCell(0, 0).getText().contains("Person 6"));
     }
 
+    @Test
+    public void gridWithFiltering_scrolledToEnd() {
+        // Open /vaadin-grid-it-demo/filtering
+        openTabAndCheckForErrors("filtering");
+        GridElement grid = $(GridElement.class).id("grid-with-filters");
+
+        // Scroll to the end of the grid
+        grid.scrollToRow(grid.getRowCount() - 1);
+        // Filter "Name" column with "100"
+        grid.findElement(By.tagName("vaadin-text-field")).sendKeys("100");
+        waitUntil(driver -> grid.getRowCount() == 1);
+
+        // Expect the one remaining row's first cell to contain
+        // text "Person 100"
+        Assert.assertEquals("Person 100", grid.getCell(0, 0).getText());
```

Comment:
When removing the fix, I'm getting a bit of a different actual value here compared to what is described in the issue. In the issue, and when testing manually, the cell seems to be empty. However when running the IT, it contains a value:
```
 GridViewIT.gridWithFiltering_scrolledToEnd:710 expected:<Person [100]> but was:<Person [489]>
```
Apart from that I can confirm that the change fixes the issue, and that the test fails when removing the fix, so maybe this is OK.

Reply:
![scrollToEnd_filter_rowsUpdated ANY_Chrome_ (com vaadin flow component grid it GridViewIT)](https://user-images.githubusercontent.com/1222264/144376664-88176f8a-5435-4a44-ad6e-8c789c84aea7.png)

In the error screenshot (as in the original issue), the row seems visually empty. The cells however do contain text as the test output shows, but the row is in `[part~='row'][loading]` state so the content has `opacity: 0`.

Reply:
Got it, thanks for the explanation.